proposals/20200909-fargate-falco-support.md

```diff
@@ -0,0 +1,156 @@
+# Falco userspace launch support
+
+## Summary
+
+This is a proposal on how to adress deployment of falco in a userspace
+ environment.
+
+There has been a rise in popularity of serverless architectures that are 
+managed and make kernel instrumentation impossible. This makes it quite hard to
+monitor the runtimes. This project is about defining a patching language/definition
+that allows falco to be deployed alongside a container in userspace.
+
+## Motivation 
+
+We want to run falco inside containers without access to kernel space.
+
+By creating a new project (Kilt) - that defines a patching/deployment procedure 
+it is possible to deploy falco in different userspace environments (k8s, cloud 
+formation, etc) 
+
+## Goals
+
+- Offer a definition format for patching/deploying falco along user containers
+- Define a patching file for falco
+- Offer automation when using k8s
+- Offer automation when using cloud formation
+
+
+## Non-Goals
+
+- Automating deployments done outside of goals
+
+## Proposal
+
+There are 3 main deployment methods when using vanilla AWS: 
+
+- *Existing containers outside of K8S* - This deployment type is the hardest to automate. We 
+will offer instructions on how to alter the containers to add falco. We can do 
+a migration tool for existing containers but this is not the goal of this
+ project. Adding `patch {}` instructions to the file actually offers a way
+towards offering this functionality as well
+
+- *K8S* - We will offer a K8S Admission Controller Mutating Webhook that will 
+alter pods adding falco according to patching file
+
+- *K8S-exec* - We will offer a way to execute `patch {}` instructions against 
+specific pods
+
+- *Cloud Formation* - We will offer a cloud formation macro to automatically 
+alter fargate task definitions in a CFN template
+
+
+
+### Components
+- *kilt-cfn* (PoC ready) - A lambda that patches aws task definitions and can
+ be registered as a macro
+- *kilt-k8s-webhook* (TBD) - A service that listens for webhooks and alters pod
+ resources as they are created
+- *kilt-k8s-exec* - CLI to patch existing k8s pods.
+
+
+## Workflow 
+### Preparation
+We prepare and publish a docker image containing falco. The contents would be 
```

Comment:
```suggestion
We prepare and publish a docker image containing Falco. The contents would be 
```
Please explain why we need to publish a new docker image and what its status will be w.r.t. [our definitions](https://github.com/falcosecurity/falco/blob/master/proposals/20200506-artifacts-scope-part-1.md#falco-project-evolution).

Reply:
The docker image currently published by the project is unsuitable to kilt:
* lua paths are wired (addressed in #1419, part of 0.27) - this assumes all lua files are in /usr/share/falco/lua
* current image has falco in /usr/bin and /usr/share - this cannot be exposed as a volume to another container - something like `/falco` or `/vendor/falco` would work.
* the current image does not have support binaries needed for injection
   * launcher - launch original entry point and falco as well
   * cwshipper - ship logs (currently only cloudwatch is supported). Needs to be configurable only from env vars
   * waitforever - no-op entry point